docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
 # SSV Specifications - Networking
 
-**Status: WIP**
+| Authors                                    | Status | Latest Revision |
+|:-------------------------------------------|:-------|:----------------|
+| [@amir-blox](https://github.com/amir-blox) | WIP    | JAN 22          |
+
+This document contains the networking specification for `SSV.Network`.
+
+## Overview
+
+- [x] [Fundamentals](#fundamentals)
+  - [x] [Stack](#stack)
+  - [x] [Transport](#transport)
+  - [x] [Messaging](#messaging)
+  - [x] [Network Peers](#network-peers)
+  - [x] [Identity](#identity)
+  - [x] [Network Discovery](#network-discovery)
+  - [x] [Peer Scoring](#peer-scoring)
+- [ ] [Wire](#wire)
+  - [x] [Consensus](#consensus-protocol)
+  - [x] [Sync](#sync-protocol)
+  - [ ] [Handshake](#handshake-protocol)
+- [x] [Networking](#networking)
+  - [x] [PubSub](#pubsub)
+  - [x] [PubSub Scoring](#pubsub-scoring)
+  - [x] [Message Scoring](#message-scoring)
+  - [x] [Connection Scoring](#connection-scoring)
+  - [x] [User Agent](#user-agent)
+  - [x] [Discovery](#discovery)
+  - [x] [Netowrk ID](#network-id)
+  - [x] [Subnets](#subnets)
+  - [x] [Peers Connectivity](#peers-connectivity)
+  - [x] [Forks](#forks)
+  - [x] [Security](#security)
+  - [x] [High Availability](#high-availability)
+
+## Fundamentals
+
+### Stack
+
+`SSV.Network` is a decentralized P2P network, consists of operator nodes grouped in multiple subnets.
+
+The networking layer is built with [Libp2p](https://libp2p.io/), 
+a modular framework for P2P networking that is used by multiple decentralized projects, including eth2.
+
+### Transport
+
+Network peers should support the following transports:
+- `TCP` is used by libp2p for setting up communication channels between peers. default port: `12000`
+- `UDP` is used for discovery purposes. default port: `13000`
+
+[go-libp2p-noise](https://github.com/libp2p/go-libp2p-noise) 
+is used to secure transport (see [noise protocol](https://noiseprotocol.org/noise.html)).
+
+Multiplexing of protocols over channels is achieved using [yamux](https://github.com/libp2p/go-libp2p-yamux) protocol.
+
+### Messaging
+
+Messages in the network are formatted with `protobuf` (NOTE: `v0` messages are encoded/decoded with JSON),
+and being transported p2p with one of the following methods:
+
+**Streams** 
+
+Streams are used for direct messages between peers.
```

Comment:
Is that the right way to do it? I'm pretty sure Prysm does that the same just maybe worth double checking